core/templates/dev/head/domain/state_card/StateCardObjectFactory.js

```diff
@@ -17,114 +17,189 @@
  * card domain objects used in the exploration player.
  */
 
-oppia.factory('StateCardObjectFactory', [function() {
-  var StateCard = function(
-      stateName, currentParams, contentHtml, interactionHtml, interaction,
-      leadsToConceptCard, destStateName, inputResponsePairs) {
-    this._stateName = stateName;
-    this._currentParams = currentParams;
-    this._contentHtml = contentHtml;
-    this._interactionHtml = interactionHtml;
-    this._leadsToConceptCard = leadsToConceptCard;
-    this._destStateName = destStateName;
-    this._inputResponsePairs = inputResponsePairs;
-    this._interaction = interaction;
-  };
-
-  StateCard.prototype.getDestStateName = function() {
-    return this._destStateName;
-  };
-
-  StateCard.prototype.getStateName = function() {
-    return this._stateName;
-  };
-
-  StateCard.prototype.getInteraction = function() {
-    return this._interaction;
-  };
-
-  StateCard.prototype.getCurrentParams = function() {
-    return this._currentParams;
-  };
-
-  StateCard.prototype.getContentHtml = function() {
-    return this._contentHtml;
-  };
-
-  StateCard.prototype.getInteractionHtml = function() {
-    return this._interactionHtml;
-  };
-
-  StateCard.prototype.getOppiaResponse = function(index) {
-    return this._inputResponsePairs[index].oppiaResponse;
-  };
-
-  StateCard.prototype.getLeadsToConceptCard = function() {
-    return this._leadsToConceptCard;
-  };
-
-  StateCard.prototype.getInputResponsePairs = function() {
-    return this._inputResponsePairs;
-  };
-
-  StateCard.prototype.getLastInputResponsePair = function() {
-    if (this._inputResponsePairs.length === 0) {
+oppia.factory('StateCardObjectFactory', [
+  'INTERACTION_SPECS', 'INTERACTION_DISPLAY_MODE_INLINE',
+  'ExplorationHtmlFormatterService', 'AudioTranslationLanguageService',
+  function(
+      INTERACTION_SPECS, INTERACTION_DISPLAY_MODE_INLINE,
+      ExplorationHtmlFormatterService, AudioTranslationLanguageService) {
+    var StateCard = function(
+        stateName, currentParams, contentHtml, interactionHtml, interaction,
+        leadsToConceptCard, destStateName, inputResponsePairs,
+        contentIdsToAudioTranslations, contentId) {
+      this._stateName = stateName;
+      this._currentParams = currentParams;
+      this._contentHtml = contentHtml;
+      this._interactionHtml = interactionHtml;
+      this._leadsToConceptCard = leadsToConceptCard;
+      this._destStateName = destStateName;
+      this._inputResponsePairs = inputResponsePairs;
+      this._interaction = interaction;
+      this._contentIdsToAudioTranslations = contentIdsToAudioTranslations;
+      this._contentId = contentId;
+    };
+
+    StateCard.prototype.getDestStateName = function() {
```

Comment:
Not sure this belongs in the StateCard. It's not displayed in the UI, and the state card is supposed to be a visual thing. Where's it used?

If it's just the frontier then it should go in the learner state service (or perhaps one of its dependencies) instead.

Reply:
It is not just used in the frontier card. The presence of a dest state name is used by the tutor card directive html to not show an interaction.
Also, I don't think we can remove it, since we can't just not show interaction for old cards and always show for the frontier card, as even in the frontier card, if a feedback is present after submitting answer (and answer leads to another card), the interaction has to be hidden, for which, this is used.

Reply:
I see, OK. But can we then replace it by something like card.isComplete()? And store a "complete" boolean on the card instead of a dest state name?

Reply:
Done, added an isCompleted function.